src/lib.rs

```diff
@@ -220,10 +191,23 @@ impl UnixStream {
     /// Connects to the socket named by `path`.
     ///
     /// This function will create a new unix socket and connect to the path
-    /// specified, performing associating the returned stream with the provided
+    /// specified, associating the returned stream with the default event loop's
+    /// handle.
+    pub fn connect<P>(p: P) -> io::Result<UnixStream>
```

Comment:
As I read this, it does not seem like it would be able to return `WouldBlock`. A socket will always be returned immediately but it might not be fully ready yet (so read/write might return `WouldBlock`).

https://docs.rs/mio-uds/0.6.4/mio_uds/struct.UnixStream.html#method.connect